js/admin.js

```diff
@@ -32,75 +39,150 @@ jQuery( document ).ready(
-		// Fills the fields based on the language dropdown list choice
-		$( '#add-lang #lang_list' ).on(
-			'selectmenuchange',
-			function() {
-				var value = $( this ).val().split( ':' );
-				var selected = $( "option:selected", this ).text().split( ' - ' );
-				$( '#lang_slug' ).val( value[0] );
-				$( '#lang_locale' ).val( value[1] );
-				$( 'input[name="rtl"]' ).val( [value[2]] );
-				$( '#lang_name' ).val( selected[0] );
+		// Callback when selectmenu widget change event is triggered.
+		var changeCallback = function( event, ui ) {
+			var value = $( event.target ).val().split( ':' );
+			var selected = $( "option:selected", event.target ).text().split( ' - ' );
+			$( '#lang_slug' ).val( value[0] );
+			$( '#lang_locale' ).val( value[1] );
+			$( 'input[name="rtl"]' ).val( [value[2]] );
+			$( '#lang_name' ).val( selected[0] );
+
+			// Refresh the flag field only if it's present.
+			if ( flagListExist ) {
 				$( '#flag_list').val( value[3] );
 
-				// Refresh the jQuery UI selectmenu flags list.
-				$( '#flag_list' ).selectmenu( 'refresh' ).trigger( 'selectmenuselect' );
+				// Refresh the jQuery UI selectmenu flag list.
+				if ( isJqueryUImin112 ) {
+					// Just a refresh of the menu is needed with jQuery UI 1.12 because _renderButtonItem is triggered and then inject correctly the flag.
+					selectmenuFlagList.refresh();
+				} else {
+					selectmenuRefreshButtonText( $( '#flag_list').selectmenu( 'refresh' ) );
+				}
 			}
-		);
+		};
+
+		// Create the jQuery UI selectmenu widget language list dropdown and return its instance.
+		var selectmenuLangListCallbacks = {};
+		// For the wizard we need a 100% width. So we override the previous defined value of selectmenuOptions. Remind that jQuery UI selectmenu widget substract 2% to this value.
+		if( $( '#lang_list' ).closest( '.pll-wizard-content' ).length > 0 ) {
+			selectmenuOptions = Object.assign( selectmenuOptions, { width: '102%' } );
+		}
+
+		// There is no need of create and select callbacks with jQuery UI 1.12 because overrinding _renderButtonItem method do the job.
+		if ( isJqueryUImin112 ) {
+			selectmenuLangListCallbacks = {
+				change: changeCallback,
+			};
+		} else {
+			selectmenuLangListCallbacks = {
+				create: createSelectCallback,
+				select: createSelectCallback,
+				change: changeCallback,
+			};
+		}
+		if ( langListExist ) {
+			var selectmenuLangList = $( '#lang_list' ).selectmenu( Object.assign( {}, selectmenuOptions, selectmenuLangListCallbacks ) ).selectmenu( 'instance' );
+			// Overrides each element in the jQuery UI selectmenu list by injecting flag image.
+			selectmenuLangList._renderItem = selectmenuRenderItem;
+			// Override the selected item rendering for jQuery UI 1.12
+			if ( isJqueryUImin112 ) {
+				selectmenuLangList._renderButtonItem = selectmenuRenderButtonItem;
+				selectmenuLangList.refresh(); // Need to refresh to take in account the button item rendering method after the selectmenu widget instanciaion.
+			}
+		}
```

Comment:
This code block is very similar to this previous one:

https://github.com/polylang/polylang/blob/3d981ccd5dc75f6c05ea8a7eee27af9fe4e37138/js/admin.js#L120-L130

I'd suggest you make it a function to which you pass the `$( '#lang_list' )` element, `selectmenuOptions` and `selectmenuLangListCallbacks` as parameters.

Reply:
Done! add a `initializeSelectmenuWidget()` function to both initialize selectmenu languages list and selectmenu flags list widgets with the same code.